### DISCUSSION TITLE
How I can enable load balancing in local?

### DISCUSSION THREAD

**Discussion by santime27** [langgenius/dify#5495] [Help] [answered]
Hello everyone, can I enable load balancing in local? and how.

**Comment by patryk20120** [comment]
What service do you want to load balance? Web is already doing that with pnpm

**Reply by santime27** [reply]
I wanna enable load balancing with LLM and balancing the model provider API calls between two or more, api credentials. Is this possible?
<img width="440" alt="model_config" src="https://github.com/langgenius/dify/assets/108313541/bd3e6484-3526-4cfc-9f93-1d060b006d2b">

**Reply by zhangever** [reply]
> I wanna enable load balancing with LLM and balancing the model provider API calls between two or more, api credentials. Is this possible? <img alt="model_config" width="440" src="https://private-user-images.githubusercontent.com/108313541/341863519-bd3e6484-3526-4cfc-9f93-1d060b006d2b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkwODEyMDAsIm5iZiI6MTcxOTA4MDkwMCwicGF0aCI6Ii8xMDgzMTM1NDEvMzQxODYzNTE5LWJkM2U2NDg0LTM1MjYtNGNmYy05ZjkzLTFkMDYwYjAwNmQyYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyMlQxODI4MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYWFkZjgwNDg2OWRmOTU5YzZlNjM2MzMwMTBiOTU5ZTIwMmJiODY4ODFhYjNlNTM3MjA2MmQzZjgwMTYyMzM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pyqN8zSpLGkllyYAnK_6GCvzGxMrv_tXof7Enp2jZaE">

This is a paid module. You can find the plan here: https://dify.ai/pricing

**Reply by santime27** [reply]
Yes, I'm absolutelly clear about this and I know that this feature is a paid feature in Cloud, but I asked about "How I can enable load balancing in local?" this means that in the source code there are some way to do this.

- I expose the db service and join in the Data Base, I can see the model provider tables and I change the bool column "load_balancing_enable".
<img width="369" alt="provider_settings" src="https://github.com/langgenius/dify/assets/108313541/ab9771b2-0cd3-419b-87e1-865db452909b">

-  This change enable the load balancing feature in Local.

<img width="445" alt="load_balancing_enable" src="https://github.com/langgenius/dify/assets/108313541/834694d7-c91b-4363-ae2a-bfc437a3f269">

But I ask about if someone did this in the source code.

**Reply by yuyongkratos** [reply]
@santime27 Which version of dify are you using?

**Reply by LockD0wn7** [reply]
> Yes, I'm absolutelly clear about this and I know that this feature is a paid feature in Cloud, but I asked about "How I can enable load balancing in local?" this means that in the source code there are some way to do this.是的，我对此非常清楚，我知道此功能是 Cloud 中的付费功能，但我问道“我如何在本地启用负载均衡”，这意味着在源代码中有一些方法可以做到这一点。
> 
> * I expose the db service and join in the Data Base, I can see the model provider tables and I change the bool column "load_balancing_enable".我公开了数据库服务并联接了数据库，我可以看到模型提供程序表，并更改了布尔列“load_balancing_enable”。
> 
> <img alt="provider_settings" width="369" src="https://private-user-images.githubusercontent.com/108313541/341991216-ab9771b2-0cd3-419b-87e1-865db452909b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE2MTk0MzksIm5iZiI6MTc1MTYxOTEzOSwicGF0aCI6Ii8xMDgzMTM1NDEvMzQxOTkxMjE2LWFiOTc3MWIyLTBjZDMtNDE5Yi04N2UxLTg2NWRiNDUyOTA5Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcwNFQwODUyMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZTc5ODIyMGIzNDZmM2QxNjVmNjUzOWI1MmM4ZGNkMGI4MzJhOGRjZDFhMGVmMGQ4ZGY4YTAzNDhhZDRlY2IxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.xzwco_m8hFQrlF02wKVX23sZOGE-bVv6QGcruTnVbS0">
> * This change enable the load balancing feature in Local.此更改将启用 Local 中的负载均衡功能。
> 
> <img alt="load_balancing_enable" width="445" src="https://private-user-images.githubusercontent.com/108313541/341991325-834694d7-c91b-4363-ae2a-bfc437a3f269.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE2MTk0MzksIm5iZiI6MTc1MTYxOTEzOSwicGF0aCI6Ii8xMDgzMTM1NDEvMzQxOTkxMzI1LTgzNDY5NGQ3LWM5MWItNDM2My1hZTJhLWJmYzQzN2EzZjI2OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcwNFQwODUyMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMGRkMWRhODczNWQ5ODU5YTk2Zjg2YWIxYjcwYmE3OWE3OWJhNDc0MDkwOGUyNWMyOTlkOTc4MjEyZWU5ZDcxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.BVmPZLwMaYFed6vq1oX9Qc-aJzecofzAoxJ1UQwNQG0">
> But I ask about if someone did this in the source code.但我问是否有人在源代码中这样做了。
Can this really enable load balancing?

**Comment by crazywoola** [comment, marked as answer]
This feature only available in our enterprise version.
